### PR TITLE
Add two passing and one failing unit tests for GPG signing

### PR DESCRIPTION
**Pull request by srosenthal-dd**
Failing test is due to not honoring key or signature expirations

### CODE REVIEW

**Comment by srosenthal-dd** [comment]
I read through the Go crypto lib code a bit and it barely tries to enforce expirations for either signatures or keys. It does provide a method `hasExpired := subkey.SelfSignature.KeyExpired(time.Now())` where you can check if a key has expired, so at least there seems to be a path to fix that. I'm not entirely sure if it's supposed to be the responsibility of the caller, or the library.

I read a bit through the https://github.com/ProtonMail/gopenpgp and it's different enough, it might assert expirations but it's hard to be sure without actually using it. Big question is, how do we assess how good a PGP library is?

File: gpg/backend_test.go [hunk 1]
```diff
@@ -48,6 +49,72 @@ func TestBackend_CRUDImportedKey(t *testing.T) {
...
+	masterName := "test"
+	testAccStepCreateKey(t, b, storage, masterName, keyData, false)
+
+	t.Run("signing with master key", func(t *testing.T) {
+		signature := testAccStepSign(t, b, storage, masterName, base64InputData)
+		testAccStepVerify(t, b, storage, masterName, map[string]interface{}{
+			"input":     base64InputData,
+			"signature": signature,
+		}, true)
+		testAccStepVerify(t, b, storage, masterName, map[string]interface{}{
```
Comment (trishankatdatadog) [inline gpg/backend_test.go:73]: You might also wish to test bad signatures, bad marshaling/hashing algorithms, etc, but we have coverage of that [elsewhere](https://github.com/trishankatdatadog/hvac-openpgp/blob/246a816225cd1fbaea2baaead54d771ecf9e458b/tests/test_transit_api.py#L179-L221)

File: gpg/backend_test.go [hunk 2]
```diff
@@ -48,6 +49,72 @@ func TestBackend_CRUDImportedKey(t *testing.T) {
...
+		testAccStepReadSubkey(t, b, storage, masterName, subkeyID)
+		signature := testAccStepSignWithSubkey(t, b, storage, masterName, subkeyID, map[string]interface{}{
+			"input":   base64InputData,
+			"expires": expireAfterSeconds,
+		})
+
+		// Sleep for long enough that the subkey and the signature *should have* expired
+		time.Sleep(time.Duration(3*expireAfterSeconds) * time.Second)
+
+		testAccStepVerify(t, b, storage, masterName, map[string]interface{}{
```
Comment (trishankatdatadog) [inline gpg/backend_test.go:114]: Great test!

**Review by trishankatdatadog — DISMISSED** [review]

**Comment by srosenthal-dd** [comment]
I marked the failing test to be skipped, assuming you don't want red builds.

**Review by trishankatdatadog — APPROVED** [review]

**Pull request merged**